### DISCUSSION TITLE
Error: 'refresh_token' on the admin login page

### DISCUSSION THREAD

**Discussion by bopetkovic404** [directus/directus#15810] [Archive]
### Describe the Bug

When I go to the admin login page I can see error in browser network tab: '"refresh_token" is required in either the JSON payload or Cookie"'.

### To Reproduce

Go to the admin page and don't log in. Open browser console and there is an error 400 Bad Request.

### Errors Shown

400 Bad Request
"refresh_token" is required in either the JSON payload or Cookie"

![image](https://user-images.githubusercontent.com/108407030/193028080-52a481a9-fa1d-4169-a3c7-308622770534.png)

### What version of Directus are you using?

9.18.1

### What version of Node.js are you using?

v15

### What database are you using?

Postgres 12

### What browser are you using?

Chrome, Firefox

### How are you deploying Directus?

Docker

**Comment by vanling** [comment]
This is the case because it is trying to log you in. Seems ok to me, maybe it should be a  http status 403? or Not try to refresh if there is no cookie with a refresh token present? Not really seeing why this is a problem, unless you are logging every error and you get flooded by this one :)

**Reply by rijkvanzanten** [reply]
> Seems ok to me, maybe it should be a http status 403?

To me 403 means "you're authenticated, but you're not allowed to do X" whereas 401 means "you're not authenticated" 🤔 

> Not try to refresh if there is no cookie with a refresh token present?

The cookie is httpOnly so JS can't tell if it exists or not (to prevent xss) 🙂 That also means that we effectively just have to try every time cause we don't know whether or not the cookie exists, which combined with the fact that we can't tell the browser that we expect/accept a 401 response means that we end up with this (harmless) error

**Reply by vanling** [reply]
Than it works exactly as it should work 👍  (and i meant 401 =P )

**Reply by rijkvanzanten** [reply]
I think 400 is appropriate for this particular case! In this case, the refresh endpoint is called without a token at all, which is a bad request, as the request requires a token to be passed. If an invalid token were to be passed, 401 is more appropriate 🙂

**Comment by azrikahar** [comment]
I'd agree with @vanling in this case, as it could be some form of the cookie was cleared on expiry (could be inactivity I assume, can't say I'm 100% sure myself 🤔). However if this is in fact a perpetual error that happens even on every single refresh/logins, feel free to let us know 👍 Moving this to discussion for now.
